### ISSUE TITLE
How can I reproduce the best result (mAP=88.2%) on Market1501?

### ISSUE THREAD

**Issue by Xiangyu-CAS** [michuanhaohao/reid-strong-baseline#133]
At first, thank you for your great job!
I was trying to reproduce your results on market1501, the configs were the same as "Experiment-all_tricks-without_center-market.sh"

| model| mAP(my results) | mAP(orignial) |configs|
| :------: |  :------: | :------: |:------: |
| r50|86.3%|85.7%| softmax_triplet.yml|
| r50-ibn-a|87.1%|88.2%|softmax_triplet.yml|

How can I reproduce the best result (mAP=88.2%)? I didn't use center loss, but it seems like it won't impact as much as  1%

**Comment by michuanhaohao** [comment]
Please train the model with center loss. Some experimental results may result in ~0.5% differences.

**Comment by Xiangyu-CAS** [comment]
@michuanhaohao : ) , thanks for quick reply. I just tried training with center loss as you said, it improved from 87.1% to 87.4%, but still has up to 0.8% margin to your claimed result

| model| mAP | configs|
| :------: |  :------: | :------: |
|r50-ibn-a|87.4%|softmax_triplet_with_center.yml |

What esle can I do?  Does pytorch version matter? My version is 1.2.0

**Comment by michuanhaohao** [comment]
I am also confused that you achieved better performance with the R50 but worse performance with R50-ibn-a. Maybe you can try to change your pytorch version. I remember the type of GPU devices may influence the performance. We used 1080ti GPUs.

**Comment by Xiangyu-CAS** [comment]
@michuanhaohao ORZ....Sadly, I deployed it on a 1080Ti computer and installed pytorch 1.0.0, but the result even droped from 87.4% to 87.2%, which could considered as  acceptable fluctuation. I swear the only thing I did was simply changing path and runing script, [my change] (https://github.com/Xiangyu-CAS/reid-strong-baseline/commit/56467b214a240475cfc0bcef83a0ebde4cd48941)

**Comment by H-YunHui** [comment]
@Xiangyu-CAS
I have the same problem as you, the highest mAP I got was 86.6, can you share the trained weights?

**Comment by Xiangyu-CAS** [comment]
@H-YunHui  [LINK](https://drive.google.com/open?id=1lm7c0V8TPacEZ3MaQKD3i6_Rf5wBwo6e)

**Comment by michuanhaohao** [comment]
Where did you download the pre-trained model of R50-a?

**Comment by Xiangyu-CAS** [comment]
@michuanhaohao I aslo had the same suspicion that pretrain model was not correctly prepared. Thus, I tried two pretrain model, one from your download link in Readme, and the other from https://github.com/XingangPan/IBN-Net.
However, there is difference....

**Comment by H-YunHui** [comment]
@Xiangyu-CAS
which pre-trained model can get the higher mAP and rank1?  Can you provide a detailed comparison?

**Comment by Xiangyu-CAS** [comment]
@H-YunHui Sorry, "there is NO difference", edited

**Comment by H-YunHui** [comment]
@Xiangyu-CAS   Thank you.

**Comment by Xiangyu-CAS** [comment]
@H-YunHui @michuanhaohao After setting triplet_loss_weight=0.5 and change init method of BN from "nn.init.constant_(m.weight, 1.0)" to "nn.init.normal_(m.weight, std=0.001)", I can obtain a result of 88.0% now (without center) 
But still, I can't reproduce your best result, even after tried 3 times on there diference machines.... 
Hope someone can help me find out the reason T_T

**Comment by Xiangyu-CAS** [comment]
@michuanhaohao BTW, do you have any results of combining local features? I saw your answer in one of the issuses, which saying you did experiments of combining local features and had recorded in your journal version paper, but I can't find where it is.
I also tried to use local feature based on your project, and finally got 89% mAP, with only 1% benefits.

**Comment by zhanghongruiupup** [comment]
> @michuanhaohao BTW, do you have any results of combining local features? I saw your answer in one of the issuses, which saying you did experiments of combining local features and had recorded in your journal version paper, but I can't find where it is.
> I also tried to use local feature based on your project, and finally got 89.0% mAP, with only 1% benefits. Can you share your idea of combining local features?

hello,This is a surprising result for 89.0% mAP,Can you share the code about local features?

**Comment by Xiangyu-CAS** [comment]
@zhanghongruiupup You can refer to this [repo](https://github.com/JDAI-CV/fast-reid/blob/master/MODEL_ZOO.md), a enhanced version of strong-baseline.

**Comment by r07942156** [comment]
@Xiangyu-CAS  Hello, could you please share the information on how you achieve 89@mAP~
